torch/_inductor/fx_utils.py

```diff
@@ -120,7 +124,7 @@ def is_aten_node(node):
                 updating_node = processing.pop()
```

Comment:
I'm pretty sure this is wrong. We're updating nodes in a depth first traversal, but that means the dependencies of the node might not have been updated yet. I think it would make more sense to merge the outer and inner loops:
```python
to_process = set()
for node in nodes:
    if self.node_hash(node) in self.processed_hashes and id(node) not in to_process:
        continue

    # ... update this node
    
    for user in node.users:
        to_process.add(id(user))
```

```

Reply:
On second though I think this is correct, but only because we potentially revisit the same node multiple times which is inefficient. IMO the single loop is probably better.